### PR TITLE
fix(sql): broaden firewall and improve error message

### PR DESCRIPTION
**Pull request by xzf0587**
1. broaden the ip range of the local firewall rule from class C(a.b.c.0-a.b.c.255) to class B(a.b.0.0-a.b.255.255)
2. improve the error message about adding firewall rule.
image

### CODE REVIEW

File: packages/fx-core/src/plugins/resource/sql/sqlClient.ts [hunk 1]
```diff
@@ -26,18 +27,22 @@ export class SqlClient {
         return false;
       }
     } catch (error) {
-      this.ctx.logProvider?.error(
-        ErrorMessage.SqlCheckDBUserError.message(this.config.identity, error.message)
-      );
-      if (error?.message?.includes(ErrorMessage.AccessMessage)) {
+      if (error?.message?.includes(ErrorMessage.AccessMessage) && extractIp(error.message)) {
+        const ip = extractIp(error.message);
```
Comment (adashen) [inline packages/fx-core/src/plugins/resource/sql/sqlClient.ts:31]: better to cache the extracted value in the previous if condition.

Comment (xzf0587) [reply packages/fx-core/src/plugins/resource/sql/sqlClient.ts]: it is fixed. Thanks.

**Review by adashen — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/sql/errors.ts [hunk 2]
```diff
@@ -91,7 +91,8 @@ export class ErrorMessage {
 
   public static readonly SqlAccessError = {
     name: "SqlAccessError",
-    message: (sqlName: string, detail = "") => `Failed to access server '${sqlName}'. ${detail}`,
+    message: (sqlName: string, ip: string, detail = "") =>
+      `Failed to access server '${sqlName}', please refer the help to add '${ip}' into the server firewall rule since your public ip may occasionally change. ${detail}`,
```
Comment (timngmsft) [inline packages/fx-core/src/plugins/resource/sql/errors.ts:95]: I don't think this needs to change. The '${detail}' contains enough information.

Comment (timngmsft) [reply packages/fx-core/src/plugins/resource/sql/errors.ts:95]: The change adds redundant information.

**Review by timngmsft — CHANGES_REQUESTED** [review]

File: packages/fx-core/src/plugins/resource/sql/errors.ts (hunk 2 above)
Comment (xzf0587) [reply packages/fx-core/src/plugins/resource/sql/errors.ts]: It is fixed. Thanks.

**Review by timngmsft — APPROVED** [review]

**Pull request merged**